tests/Unit/Evolution/Systems/Ccz4/FiniteDifference/Test_SoTimeDerivative.cpp

```diff
@@ -0,0 +1,679 @@
+          Affine{-1., 1., lower_bound[0], upper_bound[0]},
+          Affine{-1., 1., lower_bound[1], upper_bound[1]},
+          Affine{-1., 1., lower_bound[2], upper_bound[2]},
+      });
+  // set up displaced logical coords
+  const auto logical_coords =
+      TestHelpers::Ccz4::fd::detail::set_logical_coordinates(subcell_mesh);
+  const auto x = coord_map(logical_coords);
+  InverseJacobian<DataVector, SpatialDim, Frame::ElementLogical,
+                  Frame::Inertial>
+      cell_centered_logical_to_inertial_inv_jacobian{
+          subcell_mesh.number_of_grid_points(), 0.0};
+
+  for (size_t i = 0; i < SpatialDim; ++i) {
+    cell_centered_logical_to_inertial_inv_jacobian.get(i, i) =
+        2.0 / gsl::at(coords_range, i);
+  }
+
+  const Element<SpatialDim> element =
+      TestHelpers::Ccz4::fd::detail::set_element();
+
+  const DirectionalIdMap<SpatialDim, evolution::dg::subcell::GhostData>
+      all_ghost_data =
+          TestHelpers::Ccz4::fd::detail::compute_ghost_data<Frame::Inertial>(
+              subcell_mesh, x, element.neighbors(), ghost_zone_size,
+              TestHelpers::Ccz4::fd::detail::Minkowski::
+                  compute_prim_solution_for_Minkowski,
+              coords_range);
+
+  // Get system evolved variables
+  // Use the physical inertial coords
+  const auto evolved_vars = TestHelpers::Ccz4::fd::detail::Minkowski::
+      compute_prim_solution_for_Minkowski(x);
+
+  // get other parameters
+  const double f = 0.1;
+  const double kappa_1 = 0.1;
+  const double kappa_2 = 0.3;
+  const double kappa_3 = 0.4;
+
+  const DataVector used_for_size =
+      DataVector(subcell_mesh.number_of_grid_points(),
+                 std::numeric_limits<double>::signaling_NaN());
+  const auto k_0 = make_with_value<Scalar<DataVector>>(used_for_size, 0.0);
+  const auto eta = make_with_value<Scalar<DataVector>>(used_for_size, 0.0);
+
+  // put needed quantities into databox
+  using dt_variables_tag =
+      db::add_tag_prefix<::Tags::dt, Ccz4::fd::System::variables_tag>;
+
+  auto box = db::create<db::AddSimpleTags<
+      Ccz4::fd::System::variables_tag, ::Ccz4::Tags::Eta<DataVector>,
+      ::Ccz4::Tags::K0<DataVector>, ::Ccz4::Tags::GammaDriverParam,
+      ::Ccz4::Tags::Kappa1, ::Ccz4::Tags::Kappa2, ::Ccz4::Tags::Kappa3,
+      dt_variables_tag, evolution::dg::subcell::Tags::Mesh<SpatialDim>,
+      evolution::dg::subcell::fd::Tags::InverseJacobianLogicalToInertial<
+          SpatialDim>,
+      evolution::dg::subcell::Tags::GhostDataForReconstruction<SpatialDim>>>(
+      evolved_vars, eta, k_0, f, kappa_1, kappa_2, kappa_3,
+      Variables<typename dt_variables_tag::tags_list>{
+          subcell_mesh.number_of_grid_points()},
+      subcell_mesh, cell_centered_logical_to_inertial_inv_jacobian,
+      all_ghost_data);
+
+  // Check that all time derivatives are 0
+  ::Ccz4::fd::apply<SpatialDim>(make_not_null(&box));
+  const auto zero = DataVector(used_for_size.size(), 0.0);
+  for (auto& component :
```

Comment:
[optional]
```cpp
tmpl::for_each<tmpl::list<::Tags::dt<::Ccz4::Tags::ConformalMetric<DataVector, SpatialDim>>, ...>>(
   [&]<typename Tag>(tmpl::type_<Tag> /*meta*/) {
    for (...) { ... }
  });
```
I've done this and sometimes used 
```cpp
const std::string tag_name = db::tag_name<Tag>();
CAPTURE(tag_name);
```
in the lambda to record which tag failed :)

You can probably get the list from the `System` struct so you don't need to write it out!